GitUI/Plugin/PluginRegistry.cs

```diff
@@ -82,5 +86,95 @@ public static void Unregister(IGitUICommands gitUiCommands)
             Plugins.ForEach(p => p.Unregister(gitUiCommands));
             PluginsRegistered = false;
         }
+
+        /// <summary>
+        /// Ensures that all plugins under "Plugins" folder are listed in the config under
+        /// <c>configuration/runtime/ns:assemblyBinding/ns:probing</c> node.
+        /// </summary>
+        /// <returns>
+        ///   <see langword="true"/> if no changes to the config files were necessary; otherise <see langword="false"/> and the app must be restarted.
+        /// </returns>
+        public static bool EnsurePluginProbigPathsConfigured()
+        {
+            // Ensure all plugin directories are probed
+            // if not,  <probing privatePath="" /> node in the config is updated, and then we need to restart the app
+            string pluginLocation = Path.Combine(Path.GetDirectoryName(Application.ExecutablePath), "Plugins");
+            var appConfig = AppDomain.CurrentDomain.SetupInformation.ConfigurationFile;
+            return EnsurePluginProbigPathsConfigured(pluginLocation, appConfig);
+        }
+
+        private static bool EnsurePluginProbigPathsConfigured(string pluginsRootPath, string appConfig)
+        {
+            // get the name of the plugins folder
+            var pluginsFolderName = Path.GetFileName(pluginsRootPath.RemoveTrailingPathSeparator());
+            if (string.IsNullOrWhiteSpace(pluginsFolderName))
+            {
+                throw new ArgumentException("Invalid Plugins folder path", nameof(pluginsRootPath));
+            }
+
+            if (!Directory.Exists(pluginsRootPath))
+            {
+                // no plugins found, all good
+                return true;
+            }
+
+            var plugins = Directory.GetDirectories(pluginsRootPath, "*", SearchOption.TopDirectoryOnly);
+            if (plugins.Length < 1)
+            {
+                // no plugins found, all good
+                return true;
+            }
+
+            if (!File.Exists(appConfig))
+            {
+                // no application config, presume no plugins is needed
+                // the app is broken, not going to try to fix it
+                return true;
+            }
+
```

Comment:
There is a missing control to add here that makes that the `App.config` is always updated, the message box '"Plugins probing paths have been updated' is always displayed (but not restarted, I don't know why).

We need to update the list of paths in the `App.config` only if there is a missing path/